### ISSUE TITLE
Running time for PHATE in R

### ISSUE THREAD

**Issue by arielmadr** [KrishnaswamyLab/phateR#16]
Hi,

Thanks for developing PHATE, it looks very interesting. I'm running the phateR tutorial, but it is stuck on the "Calculating SVD..." step. I'm wondering if you have experienced this behavior, because when I run the python tutorial it takes less than 5 min. I installed the latest "0.2.7" release. 

Thanks so much,

ariel

**Comment by arielmadr** [comment]
Hi Scott, 

Thanks for your quick answer!. Here it is:
a) code in python
`
>>> import phate
>>> tree, branches = phate.tree.gen_dla(n_dim = 200, n_branch = 10, branch_length = 300,
...                                     rand_multiplier = 2, seed=37, sigma = 5)
>>> print(tree.shape, branches.shape)
(3000, 200) (3000,)
>>> phate_operator = phate.PHATE(n_jobs=-2)
>>> tree_phate = phate_operator.fit_transform(tree)
Calculating PHATE...
Calculating graph and diffusion operator...
Calculating PCA...
Calculated PCA in 0.19 seconds.
Calculating KNN search...
Calculated KNN search in 0.89 seconds.
Calculating affinities...
Calculated affinities in 0.16 seconds.
Calculated graph and diffusion operator in 1.26 seconds.
Calculating landmark operator...
Calculating SVD...
Calculated SVD in 0.81 seconds.
Calculating KMeans...
Calculated KMeans in 25.55 seconds.
Calculated landmark operator in 29.79 seconds.
Calculating optimal t...
Calculated optimal t in 3.13 seconds.
Automatically selected t = 11
Calculating diffusion potential...
Calculated diffusion potential in 1.97 seconds.
Calculating metric MDS...
Calculated metric MDS in 116.15 seconds.
Calculated PHATE in 152.31 seconds.
` 

b) code in R
`
>library(phateR)
>data(tree.data)
>tree.phate <- phate(tree.data$data, verbose=2)
`

c) output in R
`
> library(phateR)
Loading required package: Matrix
> #> Loading required package: Matrix
> #> Warning: package 'Matrix' was built under R version 3.4.4
> data(tree.data)
> tree.phate <- phate(tree.data$data, verbose=2)
/home/amadrigal/.local/lib/python3.5/site-packages/phate/phate.py:205: FutureWarning: potential_method is deprecated. Setting gamma to 1 to achieve log transformation.
  FutureWarning)
Set PHATE logging to DEBUG
Calculating PHATE...
Calculating graph and diffusion operator...
Set graphtools logging to DEBUG
Building knn graph with landmarks
Initializing [<class 'graphtools.graphs.kNNGraph'>, <class 'graphtools.graphs.LandmarkGraph'>] with arguments gamma='None', verbose='2', distance='euclidean', thresh='0.0001', n_pca='None', decay='10.0', n_jobs='1', initialize='True', knn='16', n_svd='100', random_state='None', n_landmark='2000', kernel_symm='+'
Set graphtools logging to DEBUG
Initializing kernel...
Calculating KNN search...
Calculated KNN search in 1.48 seconds.
Calculating affinities...
search_knn = 320; 17 remaining
radius search on 17
Calculated affinities in 0.29 seconds.
Using addition symmetrization.
Calculated graph and diffusion operator in 1.80 seconds.
Calculating landmark operator...
Calculating SVD...
`
The R code is stuck in that last step ( let it run for 15 hours and is still stuck)

Thanks so much,

Ariel

**Comment by scottgigante** [comment]
Hi Ariel,

This is very bizarre - PHATE should not take more than an hour unless you have hundreds of thousands of cells, and SVD should definitely not be taking that long. Do you know if the CPU is busy during those 15 hours? The phateR backend runs PHATE in Python, so the code bases _should_ be the same. 

If you could please run the following for me and post the output, I can try to debug.

In a console (bash/cmd):
```
which python # or which python3 if you use that command to open python
```

In R:
```
reticulate::py_config()
np <- reticulate::import("numpy")
sc <- reticulate::import("scipy")
skl <- reticulate::import("sklearn")
ph <- reticulate::import("phate")
gt <- reticulate::import("graphtools")
skl$`__version__`
np$`__version__`
sc$`__version__`
ph$`__version__`
gt$`__version__`
```

In Python:
```
import numpy as np
import scipy as sc
import sklearn as skl
import phate as ph
import graphtools as gt
skl.__version__
np.__version__
sc.__version__
ph.__version__
gt.__version__
```

Thanks and sorry for the inconvenience,
Scott

**Comment by arielmadr** [comment]
Yes, that's what I thought, it was very weird for me to have so different times between R and python. Here it is:
`
> which python 
~/miniconda3/bin/python
`

in R: 

`
> reticulate::py_config()
python:         /home/amadrigal/miniconda3/bin/python
libpython:      /home/amadrigal/miniconda3/lib/libpython3.5m.so
pythonhome:     /home/amadrigal/miniconda3:/home/amadrigal/miniconda3
version:        3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06)  [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
numpy:          /home/amadrigal/miniconda3/lib/python3.5/site-packages/numpy
numpy_version:  1.13.1
python versions found: 
 /home/amadrigal/miniconda3/envs/r-reticulate/bin/python
 /home/amadrigal/miniconda3/bin/python
 /usr/bin/python
 /usr/bin/python3
 /home/amadrigal/miniconda3/envs/bds_atac/bin/python
 /home/amadrigal/miniconda3/envs/bds_atac_py3/bin/python
 /home/amadrigal/miniconda3/envs/py36/bin/python
 /home/amadrigal/miniconda3/envs/snakemake-tutorial/bin/python
> np <- reticulate::import("numpy")
> sc <- reticulate::import("scipy")
> skl <- reticulate::import("sklearn")
> ph <- reticulate::import("phate")
> gt <- reticulate::import("graphtools")
> skl$`__version__`
[1] "0.19.1"
> np$`__version__`
[1] "1.13.1"
> sc$`__version__`
[1] "0.19.0"
> ph$`__version__`
[1] "0.2.7.4"
> gt$`__version__`
[1] "0.1.7"
`

in python:

`
>>> import numpy as np
>>> import scipy as sc
>>> import sklearn as skl
>>> import phate as ph
>>> import graphtools as gt
>>> skl.__version__
'0.19.1'
>>> np.__version__
'1.13.1'
>>> sc.__version__
'0.19.0'
>>> ph.__version__
'0.2.7.4'
>>> gt.__version__
'0.1.7'

`

btw, I'm using python 3.5.2 and R 3.4

Thanks a lot,

Ariel

**Comment by scottgigante** [comment]
Hi Ariel,

I admit it, I'm confused. My only thought is maybe the numpy / scipy versions are related, but they shouldn't be. Can you please update them with
```
pip install --upgrade --user numpy scipy
```

and then try running this code from R:
```
ph <- reticulate::import("phate")
tree <- ph$tree$gen_dla(n_dim = 200L, n_branch = 10L, branch_length = 300L,
                        rand_multiplier = 2L, seed=37L, sigma = 5L)
phateR::phate(tree[[1]])
```

**Comment by arielmadr** [comment]
hi Scott,

that did the trick!! now is running smoothly. 

For future reference, these numpy/scipy versions helped me: 

`

> np <- reticulate::import("numpy")
> sc <- reticulate::import("scipy")
> skl <- reticulate::import("sklearn")
> ph <- reticulate::import("phate")
> gt <- reticulate::import("graphtools")
> skl$`__version__`
[1] "0.19.1"
> np$`__version__`
[1] "1.14.5"
> sc$`__version__`
[1] "1.1.0"
> ph$`__version__`
[1] "0.2.7.4"
> gt$`__version__`
[1] "0.1.7"
`

Running the tutorial on the page also works neat now. Thanks so much for your help Scott!

Thanks,

Ariel
